plugins/woocommerce/client/blocks/assets/js/base/components/cart-checkout/shipping-rates-control-package/index.tsx

```diff
@@ -29,11 +27,10 @@ export const ShippingRatesControlPackage = ( {
 	showItems,
 	highlightChecked = false,
 }: PackageProps ): ReactElement => {
-	const { selectShippingRate, isSelectingRate } = useShippingData();
+	const { selectShippingRate, isSelectingRate, shippingRates } =
+		useShippingData();
 
-	const internalPackageCount = useSelect(
-		( select ) => select( cartStore )?.getCartData()?.shippingRates?.length
-	);
+	const internalPackageCount = shippingRates?.length || 1;
```

Comment:
I'm concerned about this. `internalPackageCount` makes me suspicious that `getCartData()?.shippingRates` are packages, whereas `useShippingData(). shippingRates` are the rates. Can you confirm?

We may need to test this against a plugin with packages, such as subscriptions.

Reply:
I was as concerned but they are the same thing, `useShippingData().shippingRates` just returns `cartState.shippingRates`. Second, I was concerted this is avoiding subscribing to it, but it uses useSelect, so it's the same.